src/ucp/proto/proto_select.c

```diff
@@ -570,10 +570,8 @@ ucp_proto_select_get_lane_map(ucp_worker_h worker,
     range_end = -1;
     do {
         range_start = range_end + 1;
-        if (!ucp_proto_select_elem_query(worker, select_elem, range_start,
-                                         &query_attr)) {
-            break;
-        }
+        ucp_proto_select_elem_query(worker, select_elem, range_start,
+                                    &query_attr);
```

Comment:
I'm afraid that we can have an infinite loop here.
If `ucp_proto_select_elem_query` returns 0, it won't change `query_attr`.

Reply:
it always invokes query `ucp_proto_config_query` and updates `proto_attr->max_msg_length`. I get this output now:
```
[1685629558.004973] [jazz11:160770:p-1]   +---------------------------+--------------------------------------------------------------------------------------------+
[1685629558.004980] [jazz11:160770:p-1]   | perftest intra-node cfg#0 | fetching atomic by ucp_atomic_op*(multi) into host memory from host, arg in host memory    |
[1685629558.004983] [jazz11:160770:p-1]   +---------------------------+-------------------------------------------------------------------------+------------------+
[1685629558.004985] [jazz11:160770:p-1]   |                         4 | atomic fetch                                                            | dc_mlx5/mlx5_4:1 |
[1685629558.004988] [jazz11:160770:p-1]   |                         8 | atomic fetch                                                            | dc_mlx5/mlx5_4:1 |
[1685629558.004990] [jazz11:160770:p-1]   +---------------------------+-------------------------------------------------------------------------+------------------+
[1685629558.005268] [jazz11:160770:p-1]   +---------------------------+-------------------------------------------------------------------------------------+
[1685629558.005274] [jazz11:160770:p-1]   | perftest intra-node cfg#0 | fetching atomic by ucp_atomic_op* into host memory from host, arg in host memory    |
[1685629558.005276] [jazz11:160770:p-1]   +---------------------------+------------------------------------------------------------------+------------------+
[1685629558.005278] [jazz11:160770:p-1]   |                         4 | atomic fetch                                                     | dc_mlx5/mlx5_4:1 |
[1685629558.005280] [jazz11:160770:p-1]   |                         8 | atomic fetch                                                     | dc_mlx5/mlx5_4:1 |
[1685629558.005282] [jazz11:160770:p-1]   +---------------------------+------------------------------------------------------------------+------------------+
```

Reply:
I see. Missed that `ucp_proto_select_elem_query` was also updated.